tutorials/scripting/c_sharp/diagnostics/GD0001.rst

```diff
@@ -0,0 +1,66 @@
+GD0001: Missing partial modifier on declaration of type which is a subclass of GodotObject
+==========================================================================================
+
+====================================  ======================================
+                                      Value
+====================================  ======================================
+**Rule ID**                           GD0001
+**Category**                          Usage
+**Fix is breaking or non-breaking**   Non-breaking
+**Enabled by default**                Yes
+====================================  ======================================
+
+Cause
+-----
+
+A type that derives from ``GodotObject`` is not declared partial.
+
+Rule description
+----------------
+
+Godot source generators add generated code to user-defined types to implement
+the integration with the engine. Source generators can't add generated code to
+types that aren't declared partial.
+
+.. code-block:: csharp
+
+    // The source generators can't enhance this type to work with Godot.
+    public class InvalidNode : Node { }
+
+    // The source generators can enhance this type to work with Godot.
+    public partial class ValidNode { }
+
+How to fix violations
+---------------------
+
+To fix a violation of this rule, add the ``partial`` keyword to the type
+declaration.
+
+.. tip::
```

Comment:
IMO this is not really helpful for this error as it you are going to have the boilerplate code collection of the century because you write all that generated code manually (so unlikely to ever happen) or you implement your own source generator (vastly more likely) and still have to mark that member as partial.

Is this already documented elsewhere or should this be moved to another page.

Reply:
I consider disabling source generators to be an advanced use case, so I agree we could remove the tip or maybe make it less prominent. I just though it would make sense to mention all the ways in which you can "fix" the error.

> Is this already documented elsewhere

It's not documented anywhere else, maybe it should be, but I don't really want to promote it too much since it's likely not what most users would want to do.

Reply:
I feel like this belongs on a page `Pointers for advanced usage scenarios, but you have to go source diving to really realize them` or alternatively it could just stay a thing that is mentions on a few Github issues, but otherwise can only be found in the source. tho mentioning it on the docs would make it clear that this is not just an internal thing for development.

Reply:
I decided to remove the tip and we can decide how to surface this feature in the future.